src/webserver.cpp

```diff
@@ -438,6 +438,15 @@ void WebServerResponse::write(const QVariant& body)
     mg_write(m_conn, data.constData(), data.size());
 }
 
+void WebServerResponse::writeBinary(const QByteArray& data)
+{
+    if (!m_headersSent) {
+        writeHead(m_statusCode, m_headers);
+    }
+
+    mg_write(m_conn, data.constData(), data.size());
+}
+
```

Comment:
Rather than adding this function, could you please make it so that `WebServerResponse::write` does the Right Thing when its argument `body` (which, you will note, is already a `QVariant`) receives a QByteArray instead of a QString?